theme/components/buttonsTheme.ts

```diff
@@ -0,0 +1,117 @@
+      },
+    },
+    medium: {
+      h: '2.875rem',
+      fontSize: '0.875rem',
+      borderRadius: '4rem',
+      boxSizing: 'bordered-box',
+      p: '0.75rem 1.5rem',
+      '.chakra-button__icon': {
+        marginRight: '0.75rem',
+      },
+    },
+    small: {
+      h: '2.125rem',
+      fontSize: '0.875rem',
+      borderRadius: '4.625rem',
+      p: '0.375rem 1rem',
+      '.chakra-button__icon': {
+        marginRight: '0.5rem',
+      },
+    },
+  },
+  variants: {
+    primary: props => ({
+      bgColor: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+      color: mode(NAMED_COLORS.WHITE, NAMED_COLORS.DEEP_BLUE)(props),
+      border: '0.063rem solid',
+      boxShadow: '0rem 0.313rem 0.875rem rgba(0, 0, 0, 0.07)',
+      '.chakra-button__icon': {
+        fill: mode(NAMED_COLORS.WHITE, NAMED_COLORS.DEEP_BLUE)(props),
+      },
+      ':hover': {
+        bgColor: mode(NAMED_COLORS.WHITE, NAMED_COLORS.DEEP_BLUE)(props),
+        color: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        borderColor: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        '.chakra-button__icon': {
+          fill: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        },
+        ':disabled': getDisabledStyle,
+      },
+      ':disabled': {
+        '.chakra-button__icon': {
+          fill: mode(NAMED_COLORS.PALE_GREY, NAMED_COLORS.DARKER_GREY_2)(props),
+        },
+        ...getDisabledStyle(props),
+      },
+    }),
+    secondary: props => ({
+      bgColor: mode(NAMED_COLORS.WHITE, NAMED_COLORS.DEEP_BLUE)(props),
+      color: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+      boxShadow: '0rem 0.25rem 0.688rem rgba(0, 0, 0, 0.04)',
+      border: '0.063rem solid',
+      borderColor: mode(NAMED_COLORS.LIGHTER_GREY, NAMED_COLORS.WHITE)(props),
+      '.chakra-button__icon': {
+        fill: mode(NAMED_COLORS.BLACK, NAMED_COLORS.WHITE)(props),
+      },
+      ':hover': {
+        '.chakra-button__icon': {
+          fill: mode(NAMED_COLORS.BLACK, NAMED_COLORS.BLACK)(props),
+        },
+        bgColor: NAMED_COLORS.WHITE,
+        borderColor: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        color: NAMED_COLORS.DEEP_BLUE,
+        ':disabled': getDisabledStyle,
+      },
+      ':disabled': {
+        '.chakra-button__icon': {
+          fill: mode(NAMED_COLORS.PALE_GREY, NAMED_COLORS.DARKER_GREY_2)(props),
+        },
+        ...getDisabledStyle(props),
+      },
+    }),
+    text_link: props => ({
+      color: mode(NAMED_COLORS.GREY, NAMED_COLORS.PALE_GREY)(props),
+      '.chakra-button__icon': {
+        marginRight: '0.75rem',
+        fill: mode(NAMED_COLORS.LIGHTER_GREY, NAMED_COLORS.PALE_GREY)(props),
+      },
+      ':hover': {
+        color: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        '.chakra-button__icon': {
+          fill: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        },
+      },
+    }),
+  },
+}
```

Comment:
Probably fine as is, but it looks like there might be a reusable set of definitions we could have to keep things more DRY ?

Reply:
For now it is ok. We need to create more components customization and only after that move some code in common parts.